be/src/vec/functions/array/function_array_element.h

```diff
@@ -0,0 +1,224 @@
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        DCHECK(is_array(arguments[0])) << "first argument for function: " << name << " should be DataTypeArray";
+        DCHECK(is_integer(arguments[1])) << "second argument for function: " << name << " should be Integer";
+        return make_nullable(check_and_get_data_type<DataTypeArray>(arguments[0].get())->get_nested_type());
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        NullMapBuilder builder;
+        ColumnsWithTypeAndName args;
+        auto col_left = block.get_by_position(arguments[0]);
+        if (col_left.column->is_nullable()) {
+            auto null_col = check_and_get_column<ColumnNullable>(*col_left.column);
+            builder.init_source(null_col->get_null_map_column().get_data().data());
+            args = {{null_col->get_nested_column_ptr(), remove_nullable(col_left.type), col_left.name},
+                    block.get_by_position(arguments[1])};
+        } else {
+            args = {col_left,  block.get_by_position(arguments[1])};
+        }
+        builder.init_sink(input_rows_count);
+
+        auto result_type = remove_nullable(check_and_get_data_type<DataTypeArray>(args[0].type.get())->get_nested_type());
+
+        auto res_column = perform(args, result_type, builder, input_rows_count);
+        if (!res_column) {
+            return Status::RuntimeError(fmt::format("unsupported types for function {}({}, {})", get_name(),
+                                        block.get_by_position(arguments[0]).type->get_name(),
+                                        block.get_by_position(arguments[1]).type->get_name()));
+        }
+        block.replace_by_position(result, ColumnNullable::create(std::move(res_column), std::move(builder).get_null_map_column_ptr()));
+        return Status::OK();
+    }
+
+    ColumnPtr perform(const ColumnsWithTypeAndName& arguments, const DataTypePtr& result_type, NullMapBuilder& builder, size_t input_rows_count) {
+        ColumnPtr res;
+        if (!((res = execute_number<Int8>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<UInt8>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<Int16>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<UInt16>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<Int32>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<UInt32>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<Int64>(arguments, result_type, builder, input_rows_count))
+             || (res = execute_number<UInt64>(arguments, result_type, builder, input_rows_count))
```

Comment:
why do not have float/double ？

Reply:
> 

float and double already added, other types will support in ArrayType Milestone2